fourshores-company/greenpot-backend
# implement google authentication

#### What does this PR do?
- Implements the feature to enable users to sign up/log through google's social authentication

#### Description of Task to be completed?
- Integrate both passport.js and passport-google-oauth packages
- Enable Google+ API on FourShores's google developer console, setting up the Green Pot app and acquiring the Client Id and Secret
- Setup up social authentication logic to create a new user if it doesn't exist

#### How should this be manually tested?
- In the .env file provide the GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, and GOOGLE_CALLBACK.
- The GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET are gotten from your app on the google developer console.
- Ensure that the GOOGLE_CALLBACK given in your environment file is a fully qualified URL in strings e.g 'http://localhost:3000/v1.0/api/auth/google'
- Also, make sure that the GOOGLE_CALLBACK given in the .env file exactly matches any of the authorized "redirect URIs" you would have listed on the google developer console
- Start the server by running "npm run start:dev"
- Open a browser and go to the URL, "localhost:3000/v1.0/api/auth/google"
- Enter google credentials to login/signup
